### PR TITLE
Refactor Hackbot copy system

### PR DESCRIPTION
**Pull request by paked**
TLDR:

Using `{team: team, event: event, first_name: 'Harrison'}`

```yml
# message.yml

message: Helllo! I am <@<%= info[:team].bot_user_id %>. This events timestamp is <%= info[:event] %>. The custom parameter you passed in is <%= info[:first_name] %>.
```

Using `{first_name: 'Harrison'}`. (`team` and `event` are passed in at the `Hackbot::Interaction` level)

```yml
# message.yml

message: Hello! I am <%= team.bot_mention %>. This events timestamp is <%= event[:ts] %>. The custom parameter you passed in is <%= first_name %>.
```

This PR refactors Hackbot's copy system to be a bit nicer to play with.

- We now have some nice default values in available in the copy scope. You're able to access `event` and `team` without passing them in explicitly.
- Instead of accessing values through a hash, you access them directly. This means that if a value does not exist an error will be thrown. Be aware.

I've gone through and tried out all of the conversations to make sure they work, and I'm reasonably positive they do. However, I'd appreciate it if someone could just sanity check that for me 😆

### CODE REVIEW

**Comment by paked** [comment]
Thoughts on adding in some helper functions to do things like this:

- `<%= leader(id).name %>` => `Harrison Shoebridge`
- `<%= club(id).name %>` =>`Glenaeon Rudolf Steiner School`
- `<%= club(leader_id: 25).name %>` =>`Glenaeon Rudolf Steiner School`
- `<%= mention slack_id %>` => `<@BWHDHHEE>`

Not sure what a complete list would look like.

Thoughts?

**Comment by MaxWofford** [comment]
@paked I think that's a separate change than this PR. Let's discuss in-person.
